plugin.go

```diff
@@ -117,6 +121,16 @@ func (p Plugin) Exec() error {
 
 		req.Header.Set("Content-Type", p.Config.ContentType)
 
+		if p.Config.Secret != "" {
+			// generate signature with secret and body
+			h := hmac.New(sha256.New, []byte(p.Config.Secret))
+			h.Write(b)
+			sha := hex.EncodeToString(h.Sum(nil))
+
+			// append signature to headers
+			req.Header.Set("X-Drone-Signature", sha)
```

Comment:
It would probably be nice to include signature algorithm for future extendability, ex:
```suggestion
			req.Header.Set("X-Drone-Signature", fmt.Sprintf("sha256=%s", sha))
```

Reply:
Totally agree!